rfcs/text/0009_screenshot_mode_service.md

```diff
@@ -0,0 +1,244 @@
+task that it currently has which hurt performance: wasted rendering that is
+replaced with custom styles that make the page "reportable."
+
+The technical advantage of having such as service also leads to making Kibana
+application pages "printable", in the sense that sending the page to a printer
+for a hard copy results in something more meaningful and specialized for paper
+than today's Kibana can guarantee. This isn't a big concern for Kibana since
+there isn't the expectation to improve printing Kibana, but that technical
+direction is appropriate for improving PDF report generation.
+
+# Detailed design
+
+The Screenshot Mode Service is a callable API availble in the context of a
+request on the server side, and as a pubic coreSetup API Javascript.
+
+The data provided by the Screenshot Mode are signals about:
+ - whether or not the context of the page load is for a screenshot capture
+ - layout and dimensions that the result image is expected to be
+ - time zone for which to format the page data
+
+There are a few ways for the service to obtain the data for its internal state.
+The most typical way would be to have a URL query string variable with the
+raw data and have the service read it when and API client is constructed. The
+service would provide some abstractions around that data to the different
+applications.
+
+The new query string parameter could appear in the URL of any Kibana
+application. Adding new URL query string parameters is a non-breaking change,
+and applications would not be impacted by having this additional data in the URL.
+
+Since the service data needs to be an object, the best way to encode it as data
+in a URL is to use Rison encoding. An example of what that would look like in a
+Kibana URL (with linebreaks added):
+
+```
+http://localhost:5601/app/kibana#/discover?_g=()
+&_a=(columns:!(_source),index:'6a047550-4e87-11ea-ad93-21278bc67061',interval:auto,query:(language:kuery,query:''),sort:!())
+&screenshots=(enabled:!t,layout:(height:500,width:400),timezone:PST)
+```
+
+That shows an example way of encoding this raw data into the URL as a global
+variable called `screenshots`:
+```
+{
+  "enabled": true,
+  "layout": {
+    "height": 500,
+    "width": 400
+  },
+  "timezone": "PST"
+}
+```
+
+The rest of the URL is needed by the application and is used for recreating a
+certain state of data in the page. In order to be "reportable" these parts of
+the URL must be totally sufficient for recreating everything the user expects
+to see in a report.
+
+The `screenshot` object has to be crafted in Javascript code. That can be done
+by anything, but generally it would be done in Kibana Reporting as a
+modification of the URL it is reporting on. Currently, this kind of data acts
+as job parameters for PNG and PDF Reporting jobs. Each source document for a
+Reporting job already has these fields - they are defined by the applications
+in the various points of Reporting integration.
```

Comment:
I'm not sure if I follow this paragraph or not, what's the point it's trying to make?